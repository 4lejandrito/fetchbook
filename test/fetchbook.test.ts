import { expect, test } from "bun:test";
import { $ } from "execa";

function $test(command: TemplateStringsArray) {
  test(command.toString(), async () => {
    const { exitCode, stdout, stderr } = await $(command);
    expect(exitCode).toMatchSnapshot();
    expect(stdout.replaceAll(/\s*"date": ".+",?/g, "")).toMatchSnapshot();
    expect(stderr).toMatchSnapshot();
  });
}

$test`fetchbook`;
$test`fetchbook --help`;
$test`fetchbook --version`;
$test`fetchbook run --all`;
$test`fetchbook run test/pass --all`;
$test`fetchbook run test/pass --all -v`;
$test`fetchbook run test/fail --all`;
$test`fetchbook run test --all`;
$test`fetchbook run examples --all`;
$test`fetchbook run test/pass/add-post.fetch.ts`;
$test`fetchbook run test/fail/get-posts.fetch.ts`;
$test`fetchbook export --all`;
$test`fetchbook export test/pass/add-post.fetch.ts`;
