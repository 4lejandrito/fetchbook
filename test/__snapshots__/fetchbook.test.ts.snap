// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchbook 1`] = `0`;

exports[`fetchbook 2`] = `""`;

exports[`fetchbook 3`] = `
"Usage: fetchbook [options] [command]

Manage your HTTP requests

Options:
  -V, --version             output the version number
  -h, --help                display help for command

Commands:
  init [name]               initialize a fetchbook project
  run [options] [story]     run your stories
  export [options] [story]  export your stories to existing formats
  help [command]            display help for command"
`;

exports[`fetchbook --help 1`] = `0`;

exports[`fetchbook --help 2`] = `
"Usage: fetchbook [options] [command]

Manage your HTTP requests

Options:
  -V, --version             output the version number
  -h, --help                display help for command

Commands:
  init [name]               initialize a fetchbook project
  run [options] [story]     run your stories
  export [options] [story]  export your stories to existing formats
  help [command]            display help for command"
`;

exports[`fetchbook --help 3`] = `""`;

exports[`fetchbook --version 1`] = `0`;

exports[`fetchbook --version 2`] = `"2.1.1"`;

exports[`fetchbook --version 3`] = `""`;

exports[`fetchbook run --all 1`] = `0`;

exports[`fetchbook run --all 2`] = `
"‚ùØ Examples / Pokeapi
‚ùØ GET Blastoise ...
‚úî GET Blastoise 200
‚ùØ GET Charizard ...
‚úî GET Charizard 200
‚ùØ GET Venusaur ...
‚úî GET Venusaur 200
‚úî Examples / Pokeapi
‚ùØ Examples / Zelda Fan Api
‚ùØ GET A Link to the Past ...
‚úî GET A Link to the Past 200
‚ùØ GET Ocarina of Time ...
‚úî GET Ocarina of Time 200
‚úî Examples / Zelda Fan Api
‚ùØ Test / Fail
‚ùØ GET Get all posts ...
[?25h"
`;

exports[`fetchbook run --all 3`] = `
"‚úñ GET Get all posts ... [FAILED: GET Get all posts ...]
‚úñ Test / Fail [FAILED: Test / Fail]
error: expect(received).toMatchObject(expected)

- Expected  - 1
+ Received  + 1

  Object {
    "headers": Object {
      "content-type": "application/json; charset=utf-8",
    },
-   "status": 201,
+   "status": 200,
  }"
`;

exports[`fetchbook run test/pass --all 1`] = `0`;

exports[`fetchbook run test/pass --all 2`] = `
"‚ùØ Test / Pass
‚ùØ POST Add a post ...
‚úî POST Add a post 201
‚ùØ GET Add and get several posts ...
‚ùØ Before
‚ùØ POST Add a post ...
‚úî POST Add a post 201
‚ùØ POST Add a post ...
‚úî POST Add a post 201
‚ùØ POST Add a post ...
‚úî POST Add a post 201
‚úî Before
‚úî GET Add and get several posts 200
‚ùØ GET Get a missing post ...
‚úî GET Get a missing post 404
‚ùØ GET Get a post ...
‚úî GET Get a post 200
‚ùØ GET Get all posts ...
‚úî GET Get all posts 200
‚úî Test / Pass
[?25h"
`;

exports[`fetchbook run test/pass --all 3`] = `""`;

exports[`fetchbook run test/fail --all 1`] = `0`;

exports[`fetchbook run test/fail --all 2`] = `
"‚ùØ Test / Fail
‚ùØ GET Get all posts ...
[?25h"
`;

exports[`fetchbook run test/fail --all 3`] = `
"‚úñ GET Get all posts ... [FAILED: GET Get all posts ...]
‚úñ Test / Fail [FAILED: Test / Fail]
error: expect(received).toMatchObject(expected)

- Expected  - 1
+ Received  + 1

  Object {
    "headers": Object {
      "content-type": "application/json; charset=utf-8",
    },
-   "status": 201,
+   "status": 200,
  }"
`;

exports[`fetchbook run test --all 1`] = `0`;

exports[`fetchbook run test --all 2`] = `
"‚ùØ Test / Fail
‚ùØ GET Get all posts ...
[?25h"
`;

exports[`fetchbook run test --all 3`] = `
"‚úñ GET Get all posts ... [FAILED: GET Get all posts ...]
‚úñ Test / Fail [FAILED: Test / Fail]
error: expect(received).toMatchObject(expected)

- Expected  - 1
+ Received  + 1

  Object {
    "headers": Object {
      "content-type": "application/json; charset=utf-8",
    },
-   "status": 201,
+   "status": 200,
  }"
`;

exports[`fetchbook run examples --all 1`] = `0`;

exports[`fetchbook run examples --all 2`] = `
"‚ùØ Examples / Pokeapi
‚ùØ GET Blastoise ...
‚úî GET Blastoise 200
‚ùØ GET Charizard ...
‚úî GET Charizard 200
‚ùØ GET Venusaur ...
‚úî GET Venusaur 200
‚úî Examples / Pokeapi
‚ùØ Examples / Zelda Fan Api
‚ùØ GET A Link to the Past ...
‚úî GET A Link to the Past 200
‚ùØ GET Ocarina of Time ...
‚úî GET Ocarina of Time 200
‚úî Examples / Zelda Fan Api
[?25h"
`;

exports[`fetchbook run examples --all 3`] = `""`;

exports[`fetchbook run test/pass/add-post.fetch.ts 1`] = `0`;

exports[`fetchbook run test/pass/add-post.fetch.ts 2`] = `
"‚ùØ Test / Pass
‚ùØ POST Add a post ...
‚úî POST Add a post 201
‚úî Test / Pass
[?25h"
`;

exports[`fetchbook run test/pass/add-post.fetch.ts 3`] = `""`;

exports[`fetchbook run test/fail/get-posts.fetch.ts 1`] = `0`;

exports[`fetchbook run test/fail/get-posts.fetch.ts 2`] = `
"‚ùØ Test / Fail
‚ùØ GET Get all posts ...
[?25h"
`;

exports[`fetchbook run test/fail/get-posts.fetch.ts 3`] = `
"‚úñ GET Get all posts ... [FAILED: GET Get all posts ...]
‚úñ Test / Fail [FAILED: Test / Fail]
error: expect(received).toMatchObject(expected)

- Expected  - 1
+ Received  + 1

  Object {
    "headers": Object {
      "content-type": "application/json; charset=utf-8",
    },
-   "status": 201,
+   "status": 200,
  }"
`;

exports[`fetchbook export --all 1`] = `0`;

exports[`fetchbook export --all 2`] = `
"curl 'https://zelda.fanapis.com/api/games/5f6ce9d805615a85623ec2b8' -X GET
curl 'http://localhost:3000/posts' -X POST -H "content-type: application/json" --data-binary '{"it":"works!"}'
curl 'http://localhost:3000/posts' -X POST -H "content-type: application/json" --data-binary '{"it":"works!"}'
curl 'http://localhost:3000/posts' -X POST -H "content-type: application/json" --data-binary '{"it":"works!"}'
curl 'http://localhost:3000/posts' -X POST -H "content-type: application/json" --data-binary '{"it":"works!"}'
curl 'http://localhost:3000/posts' -X GET
curl 'https://pokeapi.co/api/v2/pokemon/blastoise' -X GET
curl 'https://pokeapi.co/api/v2/pokemon/charizard' -X GET
curl 'http://localhost:3000/posts/0' -X GET
curl 'http://localhost:3000/posts/1' -X GET -H "some-header: value"
curl 'http://localhost:3000/posts' -X GET -H "some-header: value"
curl 'http://localhost:3000/posts' -X GET -H "some-header: value"
curl 'https://zelda.fanapis.com/api/games/5f6ce9d805615a85623ec2ba' -X GET
curl 'https://pokeapi.co/api/v2/pokemon/venusaur' -X GET"
`;

exports[`fetchbook export --all 3`] = `""`;

exports[`fetchbook export test/pass/add-post.fetch.ts 1`] = `0`;

exports[`fetchbook export test/pass/add-post.fetch.ts 2`] = `"curl 'http://localhost:3000/posts' -X POST -H "content-type: application/json" --data-binary '{"it":"works!"}'"`;

exports[`fetchbook export test/pass/add-post.fetch.ts 3`] = `""`;
